metrics/web/test/auth/presentation/state/auth_notifier_test.dart

```diff
@@ -235,56 +248,110 @@ void main() {
-        const invalidPassword = 'password';
-        final invalidCredentials = UserCredentialsParam(
-          email: Email(invalidEmail),
-          password: Password(invalidPassword),
-        );
+      ".passwordErrorMessage is populated when the password-related error occurred during the sign-in process",
+      () async {
+        when(signInUseCase.call(any)).thenThrow(passwordAuthException);
+
+        await authNotifier.signInWithEmailAndPassword(email, password);
+
+        expect(authNotifier.passwordErrorMessage, isNotNull);
+      },
+    );
+
+    test(
+      ".authErrorMessage is populated when GoogleSignInUseCase throws",
+      () async {
+        when(googleSignInUseCase.call()).thenThrow(authException);
+
+        await authNotifier.signInWithGoogle();
+
+        expect(authNotifier.authErrorMessage, isNotNull);
+      },
+    );
 
+    test(
+      ".signInWithEmailAndPassword() clears the authentication error message on a successful sign in",
+      () async {
         when(signInUseCase.call(invalidCredentials)).thenThrow(authException);
 
-        authNotifier.signInWithEmailAndPassword(invalidEmail, invalidPassword);
+        await authNotifier.signInWithEmailAndPassword(
+          invalidEmail,
+          invalidPassword,
+        );
 
         expect(authNotifier.authErrorMessage, isNotNull);
 
-        authNotifier.signInWithEmailAndPassword(
-            'valid_email@mail.mail', 'valid_password');
+        await authNotifier.signInWithEmailAndPassword(email, password);
 
         expect(authNotifier.authErrorMessage, isNull);
       },
     );
 
+    test(
+      ".signInWithEmailAndPassword() clears the authentication email error message on a successful sign in",
+      () async {
+        when(signInUseCase.call(invalidCredentials))
+            .thenThrow(emailAuthException);
+
+        await authNotifier.signInWithEmailAndPassword(
+            invalidEmail, invalidPassword);
+
+        expect(authNotifier.emailErrorMessage, isNotNull);
+
+        await authNotifier.signInWithEmailAndPassword(email, password);
+
+        expect(authNotifier.emailErrorMessage, isNull);
+      },
+    );
+
+    test(
+      ".signInWithEmailAndPassword() clears the authentication password error message on a successful sign in",
+      () async {
+        when(signInUseCase.call(invalidCredentials))
+            .thenThrow(passwordAuthException);
+
+        await authNotifier.signInWithEmailAndPassword(
+            invalidEmail, invalidPassword);
+
+        expect(authNotifier.passwordErrorMessage, isNotNull);
+
+        await authNotifier.signInWithEmailAndPassword(email, password);
+
+        expect(authNotifier.passwordErrorMessage, isNull);
+      },
+    );
+
     test(
       ".signInWithGoogle() clears the authentication error message on a successful sign in",
-      () {
-        authNotifier.signInWithGoogle();
+      () async {
+        await authNotifier.signInWithGoogle();
```

Comment:
Should the tests of the `signInWithGoogle` method be like a previous test? I mean should we populate the `authErrorMessage` with some error and then check if a successful login clears this message?